lib/world.rb

```diff
@@ -50,6 +50,16 @@ def debug_in_after_hook
       end
     end
 
+    def debug_stmts_on_failures
+      # Intention of method is to execute additional debug statement to analyze most tricker failures.
+      logger.info("Get the list of pods under kubeapiserver namespace")
```

Comment:
@jmekkatt good start, but these prints seems testcase specific.

Reply:
@jmekkatt From the test results, it works well. Actually, when case ran into failure, we are more concerned with abnormal nodes, operators and pods, events.
`oc get node --no-headers | grep -E 'SchedulingDisabled|NotReady'`
`oc get co | grep -v '.True.*False.*False'`
`oc get pod -A | grep -Ev 'Running|Completed'`
`oc get events -A | grep -v Normal | grep " [0-9]\{2\}m " | awk '$2 <= 15m'`

Reply:
Here a way to make the cluster abnomally, you can use it to run test case, it will cause several operators Degraded.
```
oc debug node/ip-xx-0-xxx-242.us-east-2.compute.internal
sh-4.4# ip link | grep ens
sh-4.4# cat > test.sh <<EOF
ifconfig ens3 down
sleep 300
ifconfig ens3 up
EOF
sh-4.4# bash ./test.sh &
```

Reply:
I would suggest you to use a tag specific `After` block so that this debugging only happens with the tests that need it. This is without looking in details whether this debugging is universally useful or not. Just on the outlook it appears too much to add this unconditionally.

I leave to OpenSHift QE team to decide how to do. Just I noticed this PR in my inbox and it looked interesting so decided to comment. Cheers.